engine/src/test/java/org/camunda/bpm/engine/test/logging/ProcessDataLoggingContextTest.java

```diff
@@ -0,0 +1,836 @@
+      return configuration
+          .setLogginContextActivityId("actId")
+          .setLogginContextApplicationName("appName")
+          .setLogginContextBusinessKey("busKey")
+          .setLogginContextProcessDefinitionId("defId")
+          .setLogginContextProcessInstanceId("instId")
+          .setLogginContextTenantId("tenId");
+    }
+  };
+
+  @ClassRule
+  public static ProcessEngineBootstrapRule bootstrapRuleDisableActivityIdAndDefinitionId = new ProcessEngineBootstrapRule() {
+    public ProcessEngineConfiguration configureEngine(ProcessEngineConfigurationImpl configuration) {
+      return configuration.setLogginContextActivityId(null)
+          .setLogginContextProcessDefinitionId("");
+    }
+  };
+
+  @Rule
+  public ProcessEngineRule engineRule = new ProvidedProcessEngineRule(bootstrapRule);
+  @Rule
+  public ProcessEngineRule engineRuleNoBusinessKey = new ProcessEngineRule(true);
+  @Rule
+  public ProcessEngineRule engineRuleOverride = new ProvidedProcessEngineRule(bootstrapRuleOverride);
+  @Rule
+  public ProcessEngineRule engineRuleNoBusinessKeyNoActIdNoDefId = new ProvidedProcessEngineRule(bootstrapRuleDisableActivityIdAndDefinitionId);
+
+  @Rule
+  public ProcessEngineTestRule testRule = new ProcessEngineTestRule(engineRule);
+
+  @Rule
+  public ProcessEngineLoggingRule loggingRule = new ProcessEngineLoggingRule();
+
+  private RuntimeService runtimeService;
+  private TaskService taskService;
+
+  @Before
+  public void setupServices() {
+    runtimeService = engineRule.getRuntimeService();
+    taskService = engineRule.getTaskService();
+    defaultEngineRegistered = false;
+  }
+
+  @After
+  public void resetClock() {
+    ClockUtil.reset();
+  }
+
+  @After
+  public void tearDown() {
+    if (defaultEngineRegistered) {
+      runtimeContainerDelegate.unregisterProcessEngine(engineRule.getProcessEngine());
+    }
+  }
+
+  @Test
+  @WatchLogger(loggerNames = PVM_LOGGER, level = "DEBUG")
+  public void shouldNotLogBusinessKeyIfNotConfigured() {
+    // given
+    BpmnModelInstance model = Bpmn.createExecutableProcess(PROCESS)
+        .startEvent("start")
+        .userTask("waitState")
+        .endEvent("end")
+        .done();
+    Deployment deployment = engineRuleNoBusinessKey.getRepositoryService()
+        .createDeployment()
+        .addModelInstance("boo.bpmn", model)
+        .tenantId(TENANT_ID)
+        .deploy();
+    // when
+    ProcessInstance instance = engineRuleNoBusinessKey.getRuntimeService().startProcessInstanceByKey(PROCESS, B_KEY);
+    engineRuleNoBusinessKey.getTaskService().complete(engineRuleNoBusinessKey.getTaskService().createTaskQuery().singleResult().getId());
+    // then
+    assertActivityLogsPresentWithoutBusinessKey(instance, Arrays.asList("start", "waitState", "end"));
+    // clean up
+    engineRuleNoBusinessKey.getRepositoryService().deleteDeployment(deployment.getId(), true);
```

Comment:
had to wrap the tear down in a `finally` block in the test cases, I unsuccessfully tried
- using separate `ProcessEnginTestRule`s for every engine rule and manage the deployments with them, this led to unclean database and cache errors
- using a global `deploymentId` and use the `TestHelper#deleteDeployment` in a `tearDown` method for all the engine rules (because in the tearDown I don't know which one created the deployment anymore), this led to unclean cache errors for tests using other engine rules except for the test case using the last engine rule that was cleaned in the tearDown

I am happy to elaborate on this together if you like to, I couldn't really root-cause this behavior.